pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/JavaInstanceStarter.java

```diff
@@ -158,6 +158,15 @@ public class JavaInstanceStarter implements AutoCloseable {
             required = false)
     public Boolean ignoreUnknownConfigFields = false;
 
+    @Parameter(names = "--merge_secrets_into_config_map", arity = 1,
+            description = "Whether to merge secrets into the connector's configuration. Only affects Sinks and Sources."
```

Comment:
Why this doesn't apply to Functions?

Reply:
It could apply to functions. I targeted Sources and Sinks because those are often cases where you are running a third party connector and in those cases, it is easier to merge the configs than updating the code. In the function runtime, it seems to be a well defined flow to get a secret from the function's context. That being said, we could also just fix each implementation.

I provide some direct examples of issues with connectors in this comment https://github.com/apache/pulsar/issues/20862#issuecomment-1650239074.